homeassistant/components/fing/device_tracker.py

```diff
@@ -0,0 +1,128 @@
+"""Platform for Device tracker integration."""
+
+from fing_agent_api.models import Device
+
+from homeassistant.components.device_tracker import ScannerEntity, SourceType
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers import entity_registry as er
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from . import FingConfigEntry
+from .coordinator import FingDataUpdateCoordinator
+from .utils import get_icon_from_type
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: FingConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Add sensors for passed config_entry in HA."""
+    coordinator = config_entry.runtime_data
+    tracked_devices: list[FingTrackedDevice] = []
```

Comment:
We don't want to keep track of the `Entity` objects we created after we passed them to `async_add_entities`. So we should rather use something unique like the mac address to keep track of what we have added so far. Be sure to use `set` since you can easily add and subtract them from another.

Reply:
I could use a set instead of a list; however, I’m unsure whether storing only the MAC address (or something unique) instead of the entire `Entity` objects would work. The `Entity` objects are later needed to retrieve the `entity_id` of entities that are no longer tracked by the agent, so they can be removed (lines 50-52).

Reply:
You rather need to use the unique id for that. entity_ids are renamable by users

Reply:
I've checked the removal of the entity from the entity manager section more closely. 
To refactor that piece of code and remove the storage of an `Entity` object inside a list/set I can move the entity removal inside the entity itself.
Basically, inside the `_handle_coordinator_update`, I can add an else clause to the existing `is not null` condition, and perform the removal with `er.async_get(self.hass).async_remove(entity_id=self.entity_id)`.
However, I'm not sure if remove the entity from the manager within the entity itself is considered a good practice.